### PR TITLE
fix(synthetics): add error handling to multilocation alert condition for invalid GUIDs

### PR DESCRIPTION
**Pull request by abasha1234**
# Description

This PR includes changes for error handling when an invalid monitor ID is provided for the resource `newrelic_synthetics_multilocation_alert_condition`.

Ticket : [https://new-relic.atlassian.net/browse/NR-257490](https://new-relic.atlassian.net/browse/NR-257490)

Fixes # (issue)
Issue: [https://github.com/newrelic/terraform-provider-newrelic/issues/2631](https://github.com/newrelic/terraform-provider-newrelic/issues/2631)

**Before the changes**, `getMonitorID()` fails with panic on wrong monitor ID 

Screenshot 2024-04-24 at 10 27 06

**After the changes**, `getMonitorID()`  throwing an error

Screenshot 2024-04-24 at 10 50 25

## Checklist:

Please delete options that are not relevant.

- [ ] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic/blob/main/CONTRIBUTING.md#testing) for instructions on running tests locally.

### CODE REVIEW

File: newrelic/structures_newrelic_synthetics_multilocation_alert_condition.go [hunk 1]
```diff
@@ -28,7 +28,11 @@ func expandMultiLocationSyntheticsCondition(d *schema.ResourceData) (*alerts.Mul
 
 	var entities []string
 	for _, x := range d.Get("entities").([]interface{}) {
-		entities = append(entities, getMonitorID(x.(string)))
+		monitorID, err := getMonitorID(x.(string))
+		if err != nil {
+			return &condition, err
+		}
```
Comment (pranav-new-relic) [inline newrelic/structures_newrelic_synthetics_multilocation_alert_condition.go:34]:
```suggestion
		if err != nil {
			return nil, err
		}
```

File: newrelic/structures_newrelic_synthetics_multilocation_alert_condition.go [hunk 2]
```diff
@@ -28,7 +28,12 @@ func expandMultiLocationSyntheticsCondition(d *schema.ResourceData) (*alerts.Mul
 
 	var entities []string
 	for _, x := range d.Get("entities").([]interface{}) {
-		entities = append(entities, getMonitorID(x.(string)))
+		monitorID, err := getMonitorID(x.(string))
+
+		if err != nil {
+			return nil, err
+		}
+		entities = append(entities, monitorID)
```
Comment (pranav-new-relic) [inline newrelic/structures_newrelic_synthetics_multilocation_alert_condition.go:36]: I just discovered from a test that we might need to further correct this logic to handle an error if "x" is not convertible into a string. If this is not implemented, upon adding a list of empty strings to `entities` in the Terraform Configuration, the following is seen 

```
Stack trace from the terraform-provider-newrelic plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 87 [running]:
github.com/newrelic/terraform-provider-newrelic/v2/newrelic.expandMultiLocationSyntheticsCondition(0x10?)
        /Users/x/myWork/src/github.com/terraform-provider-newrelic/newrelic/structures_newrelic_synthetics_multilocation_alert_condition.go:31 +0x368
```

Let's work on this tomorrow.

File: newrelic/resource_newrelic_synthetics_alert_condition.go [hunk 3]
```diff
@@ -100,13 +100,17 @@ func resourceNewRelicSyntheticsAlertConditionCreate(ctx context.Context, d *sche
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
 
 	condition := expandSyntheticsCondition(d, monitorID)
 
 	log.Printf("[INFO] Creating New Relic Synthetics alert condition %s", condition.Name)
 
-	condition, err := client.Alerts.CreateSyntheticsConditionWithContext(ctx, policyID, *condition)
+	condition, err = client.Alerts.CreateSyntheticsConditionWithContext(ctx, policyID, *condition)
```
Comment (pranav-new-relic) [inline newrelic/resource_newrelic_synthetics_alert_condition.go:113]: Can you please use a new variable such as `createSyntheticsConditionResponse` to replace `condition` here

File: testing/newrelic.tf [hunk 4]
```diff
@@ -8,4 +8,4 @@ terraform {
 
 provider "newrelic" {
   region = "US" # US or EU
-}
+}
```
Comment (pranav-new-relic) [inline testing/newrelic.tf:11]: please add a \n here to discard changes in this file

**Review by pranav-new-relic — APPROVED** [review]
Looks good to me 👍 great work on this @abasha1234

**Comment by abasha1234** [comment]
The PR change has been validated with the following cases

`invalid monitor_id`

Screenshot 2024-04-25 at 16 34 55


`monitor_id as empty string`

Screenshot 2024-04-25 at 16 35 42

**Comment by pranav-new-relic** [comment]
Awesome 👍

**Pull request merged**